controllers/task_controller.js

```diff
@@ -0,0 +1,275 @@
+// import http status codes
+const {
+	BAD_REQUEST,
+	NOT_AUTHORIZED,
+	FORBIDDEN,
+	NOT_FOUND,
+	NOT_ACCEPTABLE
+} = require("../utility/statusCodes");
+// import helper functions
+const {
+	sendError,
+	sendSuccess,
+	formatHtmlDate
+} = require("../utility/helpers");
+
+const ObjectId = require("mongoose").Types.ObjectId;
+
+module.exports.getTasks = async (req, res) => {
+	let { tid } = req.query;
+	let { gid } = req.params;
+	let taskIdFilter = {
+		$match: {}
+	};
+	let group = await Group.findById(gid);
+	if (!group) return sendError(res, "Group not found!!", NOT_FOUND);
+
+	if (
+		!group.heads.includes(ObjectId(req.user.id)) &&
+		!["lead", "core"].includes(req.user.role)
+	)
+		return sendError(res, "Not allowed to do that!!", FORBIDDEN);
+	if (tid) {
+		taskIdFilter["$match"] = { _id: ObjectId(tid) };
+	}
+	let tasks = await Task.aggregate([
+		taskIdFilter,
+		{
+			$lookup: {
+				from: "taskassignees",
+				let: { taskAssigneesArray: "$taskAssignees" },
+				pipeline: [
+					{
+						$match: {
+							$expr: {
+								$in: ["$_id", "$$taskAssigneesArray"]
+							}
+						}
+					},
+					{
+						$lookup: {
+							from: "users",
+							localField: "assigneeId",
+							foreignField: "_id",
+							as: "userData"
+						}
+					}
+				],
+				as: "taskAssigneeData"
+			}
+		}
+	]);
+	return sendSuccess(res, tasks);
+};
+
+module.exports.getUserTasks = async (req, res) => {
+	let { gid, uid } = req.params;
+	let tasks = await TaskAssignee.aggregate([
+		{
+			$lookup: {
+				from: "tasks",
+				localField: "_id",
+				foreignField: "taskAssignees",
+				as: "tasksData"
+			}
+		},
+		{
+			$facet: {
+				"Assigned Tasks": [
```

Comment:
Are these tasks assigned by the user or to the user?

Reply:
Assigned to the user.